src/systemctl/systemctl-show.c

```diff
@@ -1889,13 +1891,32 @@ static int show_one(
         if (r < 0)
                 return log_error_errno(r, "Failed to rewind: %s", bus_error_message(&error, r));
 
-        r = bus_message_print_all_properties(reply, print_property, arg_properties, arg_value, arg_all, &found_properties);
-        if (r < 0)
-                return bus_log_parse_error(r);
+        if (!FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF)) {
+                _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+                JsonVariant *inner;
+
+                if (arg_properties)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "--property is not supported when outputting json");
```

Comment:
This check should be done when parsing args, not here, after some real work has been done.

Reply:
implemented --property so no longer needed